sprints/dashboard/helpers.py

```diff
@@ -0,0 +1,366 @@
+        # We don't want to treat commitments from the other cell as "Unassigned".
+        if self.assignee and self.assignee.name not in cell_members:
+            self.assignee = "Other Cell"
+        self.description = getattr(issue.fields, issue_fields['Description'], '')
+        self.status = getattr(issue.fields, issue_fields['Status']).name
+        self.time_spent = getattr(issue.fields, issue_fields['Time Spent'], 0) or 0
+        self.time_estimate = getattr(issue.fields, issue_fields['Remaining Estimate'], 0) or 0
+        self.is_epic = getattr(issue.fields, issue_fields['Issue Type']).name == 'Epic'
+
+        try:
+            sprint = getattr(issue.fields, issue_fields['Sprint'])
+            if isinstance(sprint, list):
+                sprint = sprint[-1]
+            self.current_sprint = extract_sprint_id_from_str(sprint) == current_sprint_id
+        except (AttributeError, TypeError):
+            # Possible for epics
+            self.current_sprint = False
+        try:
+            self.story_points = int(getattr(issue.fields, issue_fields['Story Points'], 0))
+        except (AttributeError, TypeError):
+            self.story_points = 0
+        self.reviewer_1: JiraUser = getattr(issue.fields, issue_fields['Reviewer 1'], None)
+        # We don't want to treat commitments from the other cell as "Unassigned".
+        if self.reviewer_1 and self.reviewer_1.name not in cell_members:
+            self.reviewer_1 = "Other Cell"
+
+        self.review_time = self.calculate_review_time()
+        self.assignee_time = max(self.time_estimate - self.review_time, 0)  # We don't want negative values here
+
+    def get_bot_directive(self, pattern) -> int:
+        """Retrieves special directives placed for the Jira bot."""
+        try:
+            return int(re.search(pattern, self.description).group(1))
+        except (IndexError, AttributeError, TypeError):
+            return 0
+
+    def calculate_review_time(self) -> int:
+        """
+        Calculate time needed for the review.
+        Unless directly specified (with Jira bot directive), we're planning 2 hours for stories bigger than 3 points.
+        """
+        planned = self.get_bot_directive(SPRINT_REVIEW_DIRECTIVE)
+        if planned:
+            return planned * SECONDS_IN_HOUR
+
+        if self.story_points <= 3:
+            return SECONDS_IN_HOUR
+        return 2 * SECONDS_IN_HOUR
+
+    def get_recurring_time(self) -> int:
+        """Get required assignee time for the recurring story."""
+        planned = self.get_bot_directive(SPRINT_RECURRING_DIRECTIVE)
+        return planned * SECONDS_IN_HOUR
+
+    def get_epic_management_time(self) -> int:
+        """Get required assignee time for managing the epic."""
+        planned = self.get_bot_directive(SPRINT_EPIC_DIRECTIVE)
+        return planned * SECONDS_IN_HOUR or SPRINT_HOURS_RESERVED_FOR_EPIC_MANAGEMENT * SECONDS_IN_HOUR
+
+
+class DashboardRow:
```

Comment:
Not a big issue but, it seems like there's multiple concerns in this file.
There's a few helper functions and also some data access classes.
The helper functions as well as the data access classes will probably be used across the project at some point. Specially since we're talking about an abstraction layer towards Jira.
Why not moving these to a `libs/jira` (or something like that) module at the root of the project?
There could be a `libs.jira.utils` module for the functions and maybe group the classes in a `libs.jira.dashboard.models.[Dashboard, Issue, Row]` modules.

Reply:
@xirdneh, I reworked the placement of the classes and helper methods in 16147ce35faf8ff612e3c7452ddc5b4679e60130.
Please let me know if it looks a bit cleaner than before.
I'd like to keep Jira libs out of the root of the project, because there might be some confusion with this and the `jira` library we're using. Also we won't be using Jira connection outside of the dashboard.